be/src/storage/lake/compaction_scheduler.cpp

```diff
@@ -360,7 +368,8 @@ Status CompactionScheduler::do_compaction(std::unique_ptr<CompactionTaskContext>
         LOG(WARNING) << "Memory limit exceeded, will retry later. tablet_id=" << tablet_id << " version=" << version
                      << " txn_id=" << txn_id << " cost=" << cost << "s";
         context->progress.update(0);
-        // re-schedule the compaction task
+        // reset start time and re-schedule the compaction task
+        context->start_time.store(0, std::memory_order_relaxed);
```

Comment:
```suggestion
        context->start_time.store(::time(nullptr), std::memory_order_relaxed);
```

Reply:
this operation puts back task into queue after failure, so start time is set to 0